packages/snaps-sdk/src/structs.ts

```diff
@@ -0,0 +1,21 @@
+import type { Struct } from '@metamask/superstruct';
+import { record, refine } from '@metamask/superstruct';
+
+/**
+ * Refine a struct to be a non-empty record.
+ *
+ * @param Key - The struct for the record key.
+ * @param Value - The struct for the record value.
+ * @returns The refined struct.
+ */
+export function nonEmptyRecord<Key extends string, Value>(
```

Comment:
IMO add to `internals/structs` and export from there

Reply:
What if we need to use it in `snaps-utils` ?

Reply:
We export a number of internals for that reason: https://github.com/MetaMask/snaps/blob/main/packages/snaps-sdk/src/index.ts#L3-L14

Reply:
Moving it then :smile: